From akpm@linux-foundation.org Thu Dec 13 16:24:40 2007
From: Andrew Morton <akpm@linux-foundation.org>
Date: Thu, 13 Dec 2007 16:15:34 -0800
Subject: USB: testing driver: don't free a locked mutex
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net, akpm@linux-foundation.org, dwalker@mvista.com, matthias.kaehlcke@gmail.com, mingo@elte.hu
Message-ID: <200712140015.lBE0FYAx026319@imap1.linux-foundation.org>


From: Andrew Morton <akpm@linux-foundation.org>

Dopey thing to do and lockdep will (or should) warn.

Spotted by Daniel Walker.

Cc: Matthias Kaehlcke <matthias.kaehlcke@gmail.com>
Cc: Daniel Walker <dwalker@mvista.com>
Cc: Ingo Molnar <mingo@elte.hu>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/usbtest.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -1992,8 +1992,6 @@ static void usbtest_disconnect (struct u
 {
 	struct usbtest_dev	*dev = usb_get_intfdata (intf);
 
-	mutex_lock(&dev->lock);
-
 	usb_set_intfdata (intf, NULL);
 	dev_dbg (&intf->dev, "disconnect\n");
 	kfree (dev);
